docs/template_plugin/src/template_infer_request.cpp

```diff
@@ -65,159 +66,180 @@ void TemplateInferRequest::allocateDeviceBuffers() {
         }
-        // allocate the input blob
-        inputBlob->allocate();
-        _inputs[networkInput.first] = inputBlob;
-        if (inputBlobNCHW != inputBlob) {
-            inputBlobNCHW->allocate();
-        }
-        _inputsNCHW[networkInput.first] = inputBlobNCHW;
-    }
-}
-
-void TemplateInferRequest::allocateOutputBlobs() {
-    for (auto &networkOutput : _networkOutputs) {
-        SizeVector dims = networkOutput.second->getTensorDesc().getDims();
-        Precision precision = networkOutput.second->getPrecision();
-        Blob::Ptr outputBlob;
-
-        // allocate the output blob
-        Blob::Ptr outputBlobNCHW;
-        switch (precision) {
-        case Precision::FP32 :
-            outputBlobNCHW = outputBlob = InferenceEngine::make_shared_blob<float>({ precision, dims, networkOutput.second->getLayout() });
-            if (networkOutput.second->getLayout() == Layout::NHWC) {
-                outputBlobNCHW = InferenceEngine::make_shared_blob<float>({ precision, dims,  Layout::NCHW });
+        blob->allocate();
+        blobMap[blobData.first] = blob;
+
+        auto networkPresion = GetNetworkPrecision(blobData.first);
+        Blob::Ptr networkBlob;
+        switch (networkPresion) {
+        case ngraph::element::Type_t::f32 : {
+            if (precision == Precision::FP32) {
+                networkBlob = blob;
+            } else {
+                networkBlob = InferenceEngine::make_shared_blob<float>({Precision::FP32, dims, layout});
             }
-            break;
-        case Precision::FP16 :
-            outputBlobNCHW = outputBlob = InferenceEngine::make_shared_blob<int16_t>({ precision, dims, networkOutput.second->getLayout() });
-            if (networkOutput.second->getLayout() == Layout::NHWC) {
-                outputBlobNCHW = InferenceEngine::make_shared_blob<int16_t>({ precision, dims, Layout::NCHW });
-            }
-            break;
-        default:
-            THROW_IE_EXCEPTION << PARAMETER_MISMATCH_str << "Unsupported output precision: "
-                << precision << "! Supported precisions are: FP32, FP16";
+        } break;
+        default: THROW_IE_EXCEPTION << "Template Plugin: Unsupported network Input/Output Presision";
         }
-        // allocate the output blob
-        outputBlob->allocate();
-        _outputs[networkOutput.first] = outputBlob;
-        if (outputBlobNCHW != outputBlob) {
-            outputBlobNCHW->allocate();
+        if (blob != networkBlob) {
+            networkBlob->allocate();
         }
-        _outputsNCHW[networkOutput.first] = outputBlobNCHW;
+        networkBlobMap[blobData.first] = networkBlob;
     }
+}
 
-    if (_networkOutputs.empty() || _networkInputs.empty()) {
-        THROW_IE_EXCEPTION << "Internal error: no information about network's output/input";
-    }
+void TemplateInferRequest::allocateBlobs() {
+    _inputTensors.resize(_networkInputs.size());
+    _outputTensors.resize(_networkOutputs.size());
```

Comment:
`_outputTensors` and `_inputTensors` look like "device buffers" in current context. Should we move their creation to `allocateDeviceBuffers`?

Reply:
moved